### PR TITLE
Add labelWidget property to BottomNavigationBarItem

### LINKED ISSUE #104234: [Proposal]Support for padding around `BottomNavigationBarItem.label`
Before in BottomNavigationBar.title we can add style and padding. 

But title is already deprecated and now we need to use `selectedLabelStyle` to style the label. But what if i need to add padding to the label? what should we use? 

Before we can do this: 
```
BottomNavigationBarItem(
    title: Padding(....child: Text(...))
)
```
But i'm not sure how to do it now in label? 

Thanks!

### PR DESCRIPTION
**Pull request by bleroux**
## Description

This PR adds a `labelWidget` property to `BottomNavigationBarItem`.
Motivation : https://github.com/flutter/flutter/issues/104234#issuecomment-1138946867
Implementation inspiration : https://github.com/flutter/flutter/pull/104673

## Related Issue

Fixes https://github.com/flutter/flutter/issues/104234

## Tests

Adds 4 tests.

### CODE REVIEW

**Comment by cedvdb** [comment]
Semi related https://github.com/flutter/flutter/issues

**Comment by bleroux** [comment]
> Semi related https://github.com/flutter/flutter/issues

@cedvdb FYI, the issue number is missing from the link you shared.

**Comment by cedvdb** [comment]
thanks, fixed

File: packages/flutter/test/material/bottom_navigation_bar_test.dart [hunk 1]
```diff
@@ -144,25 +144,25 @@ void main() {
...
-                  icon: Icon(Icons.ac_unit),
-                  label: 'AC',
-                ),
-                BottomNavigationBarItem(
-                  icon: Icon(Icons.access_alarm),
-                  label: 'Alarm',
-                ),
-              ],
-            ),
+      MaterialApp(
```
Comment (pedromassango) [inline packages/flutter/test/material/bottom_navigation_bar_test.dart:147]: Why are we updating those two existing tests?

Comment (bleroux) [reply packages/flutter/test/material/bottom_navigation_bar_test.dart:147]: It is only some cleanup: previous formatting for those two tests was 4 spaces indentation.

**Comment by bleroux** [comment]
Closing this PR because it was implemented according to https://github.com/flutter/flutter/pull/104673 which was closed in favor of https://github.com/flutter/flutter/pull/109179. 
https://github.com/flutter/flutter/pull/109179 doesn't promote a `labelWidget` property.

**Pull request closed without merging**